### PR TITLE
Pinocchio proving key

### PR DESCRIPTION
**Pull request by ajgara**
Added some ideas for creating the verification key during setup.

### CODE REVIEW

File: src/circuits/qap.rs [hunk 1]
```diff
@@ -15,19 +17,47 @@ pub enum CreationError {
...
+    /// inputs associated v,w,y polynomials
+    /// mid associated polynomials
+    /// outputs associated v,w,y polynomials
     pub fn new(
         v: Vec<Polynomial>,
         w: Vec<Polynomial>,
         y: Vec<Polynomial>,
         target: Polynomial,
+        amount_of_inputs: usize,
+        amount_of_outputs: usize,
```
Comment (schouhy) [inline src/circuits/qap.rs:32]: Can we name these `number_of_inputs` and `number_of_outputs`?

File: src/pinocchio/setup.rs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    let s = tw.s;
+    let alpha_v = tw.alpha_v;
+    let alpha_w = tw.alpha_w;
+    let alpha_y = tw.alpha_y;
+    let beta = tw.beta;
+    let rv = tw.rv;
+    let rw = tw.rw;
+    let ry = tw.ry();
+
+    let g = FE::generator();
```
Comment (schouhy) [inline src/pinocchio/setup.rs:59]: Maybe `GroupType::generator`?

File: src/pinocchio/setup.rs [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            alpha_w: FE::random(),
+            alpha_y: FE::random(),
+            beta: FE::random(),
+            rv: FE::random(),
+            rw: FE::random(),
+        }
+    }
+}
+
+pub fn setup(qap: Qap, tw: ToxicWaste) -> EvaluationKey {
```
Comment (schouhy) [inline src/pinocchio/setup.rs:47]: I would rename `tw` to `toxic_waste`.

**Review by schouhy — APPROVED** [review]
Looks good to me. Left a few comments!

File: src/pinocchio/setup.rs (hunk 2 above)
Comment (MauroToscano) [reply src/pinocchio/setup.rs:59]: Yes, that seems better to understand what's going on and be able to change it later

File: src/circuits/qap.rs (hunk 1 above)
Comment (MauroToscano) [reply src/circuits/qap.rs:32]: Sure

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**